src/components/TableSortButton.tsx

```diff
@@ -0,0 +1,45 @@
+import { TableSortProps } from 'models';
+
+const TableSortButton = ({
+    TableSort,
+    setTableSort,
+    nameSort,
+}: TableSortProps) => {
+    const setUserTableSortClick = () => {
+        const newValue =
+            TableSort.name !== nameSort
+                ? 1
+                : TableSort.value === 0
+                ? 1
+                : TableSort.value === 1
+                ? 2
+                : 0;
+        setTableSort({ name: nameSort, value: newValue });
+    };
+    return (
+        <div
+            className="UserTableSortButton"
+            onClick={() => setUserTableSortClick()}>
+            <div
+                style={{
+                    color:
+                        TableSort.value === 2 && TableSort.name === nameSort
```

Comment:
consider moving this condition to be pre-calculated instead. also, I see pretty similar calculation few lines to the bottom, so maybe it will be appropriate to create a function for it

Reply:
Moved it to a separate function.